pkgs/by-name/ar/argc/package.nix

```diff
@@ -0,0 +1,64 @@
+{
+  lib,
+  rustPlatform,
+  stdenv, # Determin platforms.
+  fetchFromGitHub,
+  nixpkgsFun, # Re-evaluate Nixpkgs.
+  argc, # Get native argc for shell completion generation during cross-compilation.
+  installShellFiles,
+}:
+
+let
+  canExecuteHost = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
+in
+rustPlatform.buildRustPackage rec {
+  pname = "argc";
+  version = "1.14.0";
+
+  src = fetchFromGitHub {
+    owner = "sigoden";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-Li/K5/SLG6JuoRJDz2DQoj1Oi9LQgZWHNvtZ1HVbj88=";
+  };
+
+  cargoHash = "sha256-D1T9FWTvwKtAYoqFlR2OmLRLGWhPJ9D8J7lq/QKcBoM=";
+
+  nativeBuildInputs = [ installShellFiles ] ++ lib.optional (!canExecuteHost) argc;
+
+  postInstall = ''
+    INSTALL_SHELL_FILE_PATH="${lib.optionalString canExecuteHost ''''${!outputBin}/bin''${PATH:+:}''}''${PATH-}"
+
+    PATH="$INSTALL_SHELL_FILE_PATH" installShellCompletion --cmd argc \
+      --bash <(argc --argc-completions bash) \
+      --fish <(argc --argc-completions fish) \
+      --zsh <(argc --argc-completions zsh)
+  '';
+
+  disallowedRequisites = lib.optional (!canExecuteHost) argc.__spliced.buildHost or argc;
+
+  passthru = {
+    tests = {
+      cross = (nixpkgsFun {
+        localSystem = stdenv.buildPlatform;
+        crossSystem =
+          (if stdenv.buildPlatform.isAarch64 then "x86_64" else "aarch64")
+          + "-"
+          + stdenv.buildPlatform.parsed.kernel.name;
+      }).argc;
```

Comment:
Why not use pkgsCross?

Reply:
Which `pkgsCross` attribute should be chosen to test cross-compilation from `aarch64-linux` to `x86_64`?

Reply:
``pkgsCross.gnu64`` should be able to do that.

Also we can omit the x86_64 condition. On the correct platforms this is a noop.

Reply:
Addressed. Thanks a lot!